stratum/hal/lib/bcm/bcm_packetio_manager.cc

```diff
@@ -343,7 +343,8 @@ ::util::Status BcmPacketioManager::Shutdown() {
 
 ::util::Status BcmPacketioManager::RegisterPacketReceiveWriter(
     GoogleConfig::BcmKnetIntfPurpose purpose,
-    const std::shared_ptr<WriterInterface<::p4::v1::PacketIn>>& writer) {
+    const std::shared_ptr<WriterInterface<::p4::v1::StreamMessageResponse>>&
```

Comment:
Here is the main problem of using the generic parent message type: the packetio manager should only be able to create `PacketIn` messages, but now we have pass in the `StreamMessageResponse` type.

Reply:
We could create the StreamMessageResponse in the SwitchInterface, but that means we need another reader thread in the SwitchInterface to read the PacketIn message from the packet-IO manager (and maybe the digest manager) 🤔

Reply:
yes, that's another alternative. Creating a new channel and thread for each sub-type. Not my favorite, tbh.